fmralign/template_alignment.py

```diff
@@ -0,0 +1,353 @@
+        detrend : boolean, optional (default : None)
+            This parameter is passed to nilearn.signal.clean.
+            Please see the related documentation for details
+        target_affine: 3x3 or 4x4 matrix, optional (default : None)
+            This parameter is passed to nilearn.image.resample_img.
+            Please see the related documentation for details.
+        target_shape: 3-tuple of integers, optional (default : None)
+            This parameter is passed to nilearn.image.resample_img.
+            Please see the related documentation for details.
+        low_pass: None or float, optional (default : None)
+            This parameter is passed to nilearn.signal.clean.
+            Please see the related documentation for details.
+        high_pass: None or float, optional (default : None)
+            This parameter is passed to nilearn.signal.clean.
+            Please see the related documentation for details.
+        t_r: float, optional (default : None)
+            This parameter is passed to nilearn.signal.clean.
+            Please see the related documentation for details.
+        memory: instance of joblib.Memory or string (default : None)
+            Used to cache the masking process and results of algorithms.
+            By default, no caching is done. If a string is given, it is the
+            path to the caching directory.
+        memory_level: integer, optional (default : None)
+            Rough estimator of the amount of memory used by caching.
+            Higher value means more memory for caching.
+        n_jobs: integer, optional (default = 1)
+            The number of CPUs to use to do the computation. -1 means
+            'all CPUs', -2 'all CPUs but one', and so on.
+        parallel_backend: str, ParallelBackendBase instance, None (default: 'threading')
+            Specify the parallelization backend implementation. For more
+            informations see joblib.Parallel documentation
+        verbose: integer, optional (default = 0)
+            Indicate the level of verbosity. By default, nothing is printed.
+        '''
+        self.template = None
+        self.template_history = None
+        self.alignment_method = alignment_method
+        self.n_pieces = n_pieces
+        self.clustering_method = clustering_method
+        self.n_bags = n_bags
+        self.mask = mask
+        self.smoothing_fwhm = smoothing_fwhm
+        self.standardize = standardize
+        self.detrend = detrend
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.memory = memory
+        self.memory_level = memory_level
+        self.n_jobs = n_jobs
+        self.parallel_backend = parallel_backend
+        self.verbose = verbose
+
+    def fit(self, imgs, scale_template=False, n_iter=2, save_template=None):
```

Comment:
`scale_template`, `n_iter`, and `save_template` shall not be provided during fit but during class call.